xarray/tests/test_plot.py

```diff
@@ -267,6 +268,50 @@ def test2d_1d_2d_coordinates_contourf(self):
         a.plot.contourf(x="time", y="depth")
         a.plot.contourf(x="depth", y="time")
 
+    def test_contourf_cmap_set(self):
+        a = DataArray(easy_array((4, 4)), dims=["z", "time"])
+
+        cmap = deepcopy(mpl.cm.viridis)
+
+        pl = a.plot.contourf(cmap=deepcopy(cmap))
+
+        # check the set_bad color
+        assert pl.cmap._rgba_bad == cmap._rgba_bad
```

Comment:
let's do `assert pl.cmap(0) == cmap(0)` and so on. That way we are not relying on private attributes.

Reply:
I think that may actually fix (some of?) the current test failures - apparently the private attributes can sometimes be `None`, and the test fails in that case at the moment.